app/components/QuestionComment/QuestionComment.jsx

```diff
@@ -0,0 +1,373 @@
+import {useState} from 'react';
+import { VscTriangleUp, VscTriangleDown } from 'react-icons/vsc';
+import { BsThreeDotsVertical, BsCheckCircle, BsCircle } from 'react-icons/bs';
+import PropTypes from 'prop-types';
+import * as Styled from './QuestionComment.styled';
+import CounterButton from '../CounterButton';
+import QuestionerResponderInfo from '../QuestionResponderInfo';
+import Button from '~/components/Atoms/Button';
+import editIconUnselected from '~/images/ic_edit.svg';
+import deleteIconUnselected from '~/images/ic_delete.svg';
+import editIconSelected from '~/images/ic_edit_selected.svg';
+import deleteIconSelected from '~/images/ic_delete_selected.svg';
+import { getTimeDiff } from '~/utils/timeOperations';
+import useClickOutside from '~/utils/hooks/useClickOutside';
+import  {DANGER_BUTTON, DISABLED_BUTTON, SECONDARY_BUTTON, COMMENT_EDIT_PLACEHOLDER, COMMUNITY_ANSWER_TAG_TEXT} from '~/utils/constants';
+import MarkdownLinkRenderer from '~/components/MarkdownLinkRenderer';
+import CommentInputText from '~/components/CommentInput/CommentInputText';
+import { ACTIONS } from '~/utils/actions';
+import { useSubmit, useSearchParams } from '@remix-run/react';
+import { useUser } from '~/utils/hooks/useUser';
+
+function QuestionComment({ commentData, onSubmitSuccess, ...props }) {
+    const submit = useSubmit();
+    const profile = useUser();
+    const [isEditing, setIsEditing] = useState(false);
+    const [isDeleting, setIsDeleting] = useState(false);
+    const [commentHasChanged, setCommentHasChanged] = useState(false);
+    const [editIcon, setEditIcon] = useState(editIconUnselected);
+    const [deleteIcon, setDeleteIcon] = useState(deleteIconUnselected);
+    const [updatedComment, setUpdatedComment] = useState('');
+    const [upVote, setUpVote] = useState(
+      commentData.votes === undefined ? 0 : Number(commentData.votes),
+    );
+    const [upVoteActive, setUpVoteActive] = useState(
+      commentData.has_upvoted === undefined ? false : commentData.has_upvoted,
+    );
+    const [downVoteActive, setDownVoteActive] = useState(
+      commentData.has_downvoted === undefined ? false : commentData.has_downvoted,
+    );
+    const [searchParams] = useSearchParams();
+    const isAnswer = props.hasCommentAsAnswer;
+  
+    const {
+      state: showCommentOptionsState,
+      setState: setShowCommentOptions,
+      wrapperRef,
+    } = useClickOutside(isEditing || isDeleting);
+    
+    async function voteComment (comment_id, value){
```

Comment:
async is necessary? I don't see any await inside this function.

Reply:
It was before migration to Remix but think its not necesary anymore, good spot

Reply:
Done! Thanks